python/ray/dag/context.py

```diff
@@ -8,6 +8,8 @@
 _default_context: "Optional[DAGContext]" = None
 _context_lock = threading.Lock()
 
+DEFAULT_EXECUTION_TIMEOUT_S = int(os.environ.get("RAY_DAG_execution_timeout", 10))
```

Comment:
I am thinking of whether 10 seconds by default is too short.

Reply:
We don't want to have a too long timeout either, since the user would not get a signal quick enough if it hangs. What's your thought on a better default?

Reply:
Chatted offline: we decide to use 10s for now. It's OK to increase the value in the future which is not a breaking change.